src/sage/crypto/key_exchange/diffie_hellman.py

```diff
@@ -0,0 +1,341 @@
+r"""
+Diffie-Hellman Key Exchange Scheme
+
+This module contains a toy implementation of the Diffie-Hellman key exchange
+scheme.
+
+AUTHORS:
+
+- Vincent Macri (2024-07-17): initial version
+"""
+# ****************************************************************************
+#       Copyright (C) 2024 Vincent Macri <vincent.macri@ucalgary.ca>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#                  https://www.gnu.org/licenses/
+# ****************************************************************************
+
+from sage.misc.superseded import experimental
+
+from sage.crypto.key_exchange.key_exchange import KeyExchangeScheme
+
+from sage.arith.misc import is_prime
+from sage.misc.prandom import randint
+from sage.rings.integer import Integer
+from sage.rings.finite_rings.finite_field_constructor import GF
+from sage.rings.finite_rings.finite_field_prime_modn import \
+    FiniteField_prime_modn
+from sage.rings.finite_rings.integer_mod import IntegerMod_abstract
+from sage.structure.proof.proof import WithProof
+
+from typing import Union
+
+class DiffieHellman(KeyExchangeScheme):
+
+    @experimental(37305)
+    def __init__(self, p: Integer, g: Union[Integer, IntegerMod_abstract],
+                 proof: bool = True) -> None:
+        """
+        Create an instance of the Diffie-Hellman key exchange scheme using the
+        given prime ``p`` and base ``g``.
+
+        INPUT:
+
+        - ``p`` -- prime integer defining the field `\\GF{p}` that the key
+          exchanges will be performed over, must be at least 5
+
+        - ``g`` -- base for the key exchange, (coerceable to) an element of
+          `\\GF{p}` from `2` to `p - 2`
+
+        - ``proof`` -- (default: ``True``) whether to require a proof that
+          ``p`` is prime. If ``False``, a probabilistic test can be used for
+          checking that ``p`` is prime. This should be set to ``False``
+          when using large (cryptographic size) primes, otherwise checking
+          primality will take too long.
+
+        .. WARNING::
+
+            This is a toy implementation for educational use only! Do not use
+            this implementation, or any cryptographic features of Sage, in any
+            setting where security is needed!
+
+        REFERENCES:
+
+        For more information, see [PP2010]_, section 8.1.
```

Comment:
```suggestion
        For more information, see Section 8.1 of [PP2010]_.
```
Following other examples :D